src/components/DocumentRenderer/DocumentDetailsSheet.tsx

```diff
@@ -231,6 +237,30 @@ export const DocumentDetailsSheet: FunctionComponent<DocumentDetailsSheet> = ({
   const isDocumentValid = overallValidity === CheckStatus.VALID;
   const isDocumentInvalid = overallValidity === CheckStatus.INVALID;
 
+  const [ttlDescription, setTtlDescription] = useState<string | undefined>(
+    undefined
+  );
+  const hasExpired = ttl && ttl < Date.now(); //not expired until ttl value is retrieved
```

Comment:
This won't work since `ttl` only changes once when `generateQr` is called. `hasExpired` will therefore always be `false`.

We'll need to figure out when the ttl has actually reduced to 0.